Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -112,12 +112,15 @@ struct PathHierarchy {
             var nodes: [String: Node] = [:]
             nodes.reserveCapacity(graph.symbols.count)
             for (id, symbol) in graph.symbols {
-                if let existingNode = allNodes[id]?.first(where: {
-                    // If both identifiers are in the same language, they are the same symbol.
-                    $0.symbol!.identifier.interfaceLanguage == symbol.identifier.interfaceLanguage
-                    // If both have the same path components and kind, their differences don't matter for link resolution purposes.
-                    || ($0.symbol!.pathComponents == symbol.pathComponents && $0.symbol!.kind.identifier == symbol.kind.identifier)
```

Comment:
This `||` expression sometimes find the language representation of a symbol that's different from the language of the current symbol graph file. This in turn results in cloning a language specific symbol that's in the "wrong" language which removes the _only_ language from that node. Later, when computing the paths for all symbols, the cloned node without any language is skipped, resulting in a symbol that never receives a reference.